src/core/model/Stroke.cpp

```diff
@@ -72,6 +78,68 @@ auto Stroke::cloneStroke() const -> Stroke* {
 
 auto Stroke::clone() -> Element* { return this->cloneStroke(); }
 
+std::unique_ptr<Stroke> Stroke::cloneSection(const PathParameter& lowerBound, const PathParameter& upperBound) const {
+    auto s = std::make_unique<Stroke>();
+    s->applyStyleFrom(this);
+    s->width = this->width;  // stroke width, not bounding box width
+
+    if (upperBound.index == lowerBound.index) {
+        const Point& p = this->getPoint(lowerBound);
+        const Point& q = this->getPoint(upperBound);
+        s->points = {p, q};
+    } else {
+        s->points.reserve(upperBound.index - lowerBound.index + 2);
+
+        // First point
+        s->points.emplace_back(this->getPoint(lowerBound));
+
+        // end point of segment of index lowerBound.index
+        auto it = this->points.cbegin() + (std::ptrdiff_t)lowerBound.index + 1;
+
+        // end point of segment of index upperBound.index
+        auto endIt = this->points.cbegin() + (std::ptrdiff_t)upperBound.index + 1;
+
+        std::copy(it, endIt, std::back_inserter(s->points));
+
+        // Last point
```

Comment:
I think, this is over explaining:

```suggestion
        s->points.reserve(upperBound.index - lowerBound.index + 2);
        s->points.emplace_back(this->getPoint(lowerBound));
        auto const begi = std::next(begin(this->points, std::ptrdiff_t(lowerBound.index + 1));
        auto const endi= std::next(begin(this->points, std::ptrdiff_t(upperBound.index + 1));
        std::copy(begi, endi, std::back_inserter(s->points));
```